tests/data_generation/helper/analytics.py

```
@@ -0,0 +1,183 @@
+        )
+
+        # Process each outcome separately
+        if simulation_config.outcomes:
+            print("\nOUTCOME RESULTS:")
+
+            for outcome_cfg in simulation_config.outcomes.values():
+                outcome_code = outcome_cfg.code
+
+                # Count subjects with this outcome
+                outcome_subjects = df.groupby("subject_id")["code"].apply(
+                    lambda codes: (codes == outcome_code).any()
```

Ruff (0.11.9)

Function definition does not bind loop variable `outcome_code`

(B023)